pkg/adapter/adapter.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2019 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+		http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package adapter implements a sample receive adapter that generates events
+// at a regular interval.
+package adapter
+
+import (
+	"context"
+	"net/url"
+	"time"
+
+	"github.com/cloudevents/sdk-go/pkg/cloudevents"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/client"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/types"
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+	"knative.dev/eventing/pkg/adapter"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/source"
+)
+
+type envConfig struct {
+	// Include the standard adapter.EnvConfig used by all adapters.
+	adapter.EnvConfig
+
+	// Interval between events, for example "5s", "100ms"
+	Interval time.Duration `envconfig:"INTERVAL" required:"true"`
+}
+
+func NewEnv() adapter.EnvConfigAccessor { return &envConfig{} }
+
+// Adapter generates events at a regular interval.
+type Adapter struct {
+	logger   *zap.Logger
+	interval time.Duration
+	nextID   int
+	sink     client.Client
+}
+
+type dataExample struct {
+	Sequence  int
+	Heartbeat string
+}
+
+var sourceURI = types.URIRef{URL: url.URL{Scheme: "http", Host: "heartbeat.example.com", Path: "/heartbeat-source"}}
+
+func strptr(s string) *string { return &s }
+
+func (a *Adapter) newEvent() cloudevents.Event {
+
+	e := cloudevents.Event{
+		Context: cloudevents.EventContextV1{
+			ID:              uuid.New().String(),
+			Type:            "com.example.heartbeat",
+			Source:          sourceURI,
+			Time:            &types.Timestamp{Time: time.Now()},
+			DataContentType: strptr("application/json"),
+		}.AsV01(),
```

Comment:
Why is this being converted to v01 here?

Reply:
Originally it was `.AsV1()` but that wasn't working with the `gcr.io/knative-releases/github.com/knative/eventing-sources/cmd/event_display` image.  I assume I should probably find an updated `event_display` image to work with cloudevent v1 spec?

Reply:
@n3wscott thoughts here??

Reply:
Aha.  When I use a non-ancient version of the event_display image, it works with `.AsV1` my mistake for using an `eventing-sources` era image.

Fixed in 5b80d6b